src/hooks/use-authentication/index.ts

```diff
@@ -43,7 +44,7 @@ const useAuthentication = (
 	const { data, status } = useSession({
 		required: isRequired,
 		onUnauthenticated,
-	})
+	}) as { data: SessionData; status: SessionStatus }
```

Comment:
Is there a better way to do this? The `useSession` type is kinda nuts in the next auth code.

Reply:
Ty for doing this digging and finding those docs. I updated things to use the modules and it seems to work _much_ better.